### ISSUE TITLE
lxc service does not start with upstart

### ISSUE THREAD

**Issue by nickandrew** [lxc/lxc#903]
I have a bunch of containers running under an Ubuntu 14.04.4 Trusty host. My problems started when I upgraded one container to Ubuntu 15.10 Wily. My upgraded container could not start.

I searched and found #685 which indicated I should upgrade lxc via PPA. So I upgraded to 2.0.0~rc11-0ubuntu1~ubuntu14.04.1~ppa1, but now no containers start at boot time.

After some fiddling it seems that lxcfs is stopping upstart from properly starting lxc containers:

```
# initctl list | grep lxc
lxc start/starting
lxc-instance stop/waiting
lxc-net start/running
network-interface (lxcbr0) start/running
network-interface-security (network-interface/lxcbr0) start/running
lxcfs start/running, process 24810
```

I think I have been able to replicate this in a fresh container, as follows:

```
host# lxc-create -n lxctest2 -t ubuntu -- --release trusty
host# echo 'lxc.include = /usr/share/lxc/config/nesting.conf' >> /var/lib/lxc/lxctest2/config
host# echo '/var/cache/lxc var/cache/lxc none bind,create=dir' >> /var/lib/lxc/lxctest2/fstab
host# lxc-start -d -n lxctest2
host# lxc-attach -n lxctest2
lxctest2# apt-get -y install lxc software-properties-common
lxctest2# lxc-create -n lxctest2v -t ubuntu -- --release vivid
lxctest2# lxc-start -d -n lxctest2v
lxctest2# lxc-attach -n lxctest2v
lxctest2v# ps axfww
lxctest2v# # === note nothing running under init (systemd) ===
lxctest2v# exit
lxctest2# lxc-stop -n lxctest2v
lxctest2# apt-add-repository ppa:ubuntu-lxc/stable
lxctest2# apt-get update
lxctest2# initctl list | grep lxc
lxc start/running
[...]
lxctest2# COLUMNS=200 dpkg --list lxc\*
ii  lxc                                          1.0.8-0ubuntu0.3            amd64                       Linux Containers userspace tools
ii  lxc-templates                                1.0.8-0ubuntu0.3            amd64                       Linux Containers userspace tools (templates)
lxctest2# apt-get -y install lxc lxc-templates lxcfs
lxctest2# mknod /dev/fuse c 10 229
lxctest2# stop lxc
lxctest2# COLUMNS=300 dpkg --list lxc\*
ii  lxc                                                                   2.0.0~rc11-0ubuntu1~ubuntu14.04.1~ppa1  all                                     Transitional package for lxc1
ii  lxc-common                                                            2.0.0~rc11-0ubuntu1~ubuntu14.04.1~ppa1  amd64                                   Linux Containers userspace tools (common tools)
ii  lxc-templates                                                         2.0.0~rc11-0ubuntu1~ubuntu14.04.1~ppa1  amd64                                   Linux Containers userspace tools (templates)
ii  lxc1                                                                  2.0.0~rc11-0ubuntu1~ubuntu14.04.1~ppa1  amd64                                   Linux Containers userspace tools
ii  lxcfs                                                                 2.0.0~rc6-0ubuntu2~ubuntu14.04.1~ppa1   amd64                                   FUSE based filesystem for LXC
un  lxcguest                                                              <none>                                  <none>                                  (no description available)
un  lxctl                                                                 <none>                                  <none>                                  (no description available)
lxctest2# start lxcfs
lxctest2# start lxc
[ ... hangs ... ] ^C
lxctest2# initctl list | grep lxc
lxc start/starting
lxcfs start/running, process 24810
```

**Comment by stgraber** [comment]
Can you paste the "ps fauxww" output at the time of the hang?

**Comment by stgraber** [comment]
Oh and for lxcfs to work, you need it installed on the host and at every level (when nesting).

**Comment by nickandrew** [comment]
```
root@lxctest2:/# ps fauxww
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root     25554  0.0  0.0  21068  1996 ?        S    08:11   0:00 /bin/bash
root     25565  0.0  0.0  18440  1272 ?        R+   08:11   0:00  \_ ps fauxww
root     25534  0.0  0.0  21068  2128 ?        S    08:10   0:00 /bin/bash
root     25564  0.0  0.0  18168  1176 ?        S+   08:11   0:00  \_ start lxc
root         1  0.0  0.0  33244  2620 ?        Ss   Mar20   0:00 /sbin/init
root       123  0.0  0.0  15256   624 ?        S    Mar20   0:00 upstart-socket-bridge --daemon
root       202  0.0  0.0  19472   648 ?        S    Mar20   0:00 upstart-udev-bridge --daemon
root       245  0.0  0.0  15272   632 ?        S    Mar20   0:00 upstart-file-bridge --daemon
root       247  0.0  0.0  49248  1384 ?        Ss   Mar20   0:00 /lib/systemd/systemd-udevd --daemon
syslog     264  0.0  0.0 255840  1132 ?        Ssl  Mar20   0:00 rsyslogd
root       297  0.0  0.0  10228  2892 ?        Ss   Mar20   0:00 dhclient -1 -v -pf /run/dhclient.eth0.pid -lf /var/lib/dhcp/dhclient.eth0.leases eth0
root       365  0.0  0.0  12784   872 lxc/tty4 Ss+  Mar20   0:00 /sbin/getty -8 38400 tty4
root       368  0.0  0.0  12784   864 lxc/tty2 Ss+  Mar20   0:00 /sbin/getty -8 38400 tty2
root       369  0.0  0.0  12784   864 lxc/tty3 Ss+  Mar20   0:00 /sbin/getty -8 38400 tty3
root       385  0.0  0.0  61376  3040 ?        Ss   Mar20   0:00 /usr/sbin/sshd -D
root       401  0.0  0.0  23652   920 ?        Ss   Mar20   0:00 cron
root       407  0.0  0.0  12784   864 lxc/console Ss+ Mar20   0:00 /sbin/getty -8 38400 console
root       409  0.0  0.0  12784   872 lxc/tty1 Ss+  Mar20   0:00 /sbin/getty -8 38400 tty1
root      1470  0.0  0.0  23660  1444 ?        Ss   Mar20   0:00 /sbin/cgmanager --sigstop -m name=systemd
root     25548  0.0  0.0  95324   980 ?        Ssl  08:10   0:00 /usr/bin/lxcfs /var/lib/lxcfs

root@lxctest2:/# initctl list | grep lxc
lxc start/starting
lxc-instance stop/waiting
lxc-net stop/waiting
lxcfs start/running, process 25548
```

**Comment by nickandrew** [comment]
What you said indicates that lxcfs is optional, so I'll happily uninstall it. It just seems like a bug that lxcfs is somehow preventing lxc from being started by upstart.

As far as I can see, the lxc pre-start script is not being run. I don't know enough about upstart to say why.

Also if the kernel version is relevant:

```
Linux lxctest2 3.13.0-79-generic #123-Ubuntu SMP Fri Feb 19 14:27:58 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by stgraber** [comment]
yeah, we have a bug in the lxcfs packaging which @hallyn is looking at (for another bug report against lxcfs). Note however that while lxcfs IS optional as far as LXC is concerned, it very much ISN'T optional if you wish to run a container that's using systemd as it's init system.

**Comment by nickandrew** [comment]
Oh, okay :grimacing: Standing by.

**Comment by theredcat** [comment]
@stgraber @hallyn I can confirm that commit lxc/lxcfs/3eaacb725aadaa635d86cea48beda1731c6acea6 fixes this. Do you plan to tag it soon ?

**Comment by hallyn** [comment]
Sorry - what commit is that?  I cna't find a 3eaacb725aadaa in lxcfs.

**Comment by theredcat** [comment]
This one => https://github.com/lxc/lxcfs/commit/3eaacb725aadaa635d86cea48beda1731c6acea6

**Comment by Napsty** [comment]
I just came across this issue today and it's still happening. 
LXC Host: Ubuntu 18.04 Bionic, LXC 3.0.0-0ubuntu2
LXC Container: Ubuntu 14.04 Trusty, upstart 1.12.1-0ubuntu4.2

The container starts up but the services are not started. ps auxf inside container:

```
# ps auxf
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root      1546  0.0  0.0  21316  3964 ?        Ss   12:14   0:00 /bin/bash
root      1662  0.0  0.0  18456  2564 ?        R+   12:14   0:00  \_ ps auxf
root         1  0.0  0.0  33132  3764 ?        Ss   12:08   0:00 /sbin/init
root        40  0.0  0.0  36904  2472 ?        S    12:08   0:00 @sbin/plymouthd --mode=boot --attach-to-session
root        44  0.0  0.0  19696  2424 ?        Ss   12:08   0:00 plymouth-upstart-bridge
root        49  0.0  0.0  28944  2840 ?        S    12:08   0:00 mountall --daemon
root       398  0.0  0.0  15272   212 ?        S    12:08   0:00 upstart-socket-bridge --daemon
```

Had to manually start the services inside the container using /etc/init.d/ scripts.

**Comment by Blub** [comment]
The symptoms look like the issue fixed by #2787, but the responsible commit wasn't in 3.0.0, which is strange?

**Comment by brauner** [comment]
This issue should be fixed by the commit pointed to by @Blub and will be included in the next point release.
